scripts/install_python_dev_dependencies_test.py

```diff
@@ -0,0 +1,308 @@
+
+from typing import Any, Dict, List, Tuple
+
+
+class InstallPythonDevDependenciesTests(test_utils.GenericTestBase):
+
+    def test_assert_in_venv_passes_when_in_venv(self) -> None:
+        prefix_swap = self.swap(
+            sys, 'prefix', '/home/user/.pyenv/versions/3.7.10')
+        base_prefix_swap = self.swap(
+            sys, 'base_prefix', '/home/user/.pyenv/versions/oppia')
+        real_prefix_swap = self.swap(sys, 'real_prefix', '')
+        environ_swap = self.swap(os, 'environ', {})
+        with prefix_swap, base_prefix_swap, real_prefix_swap, environ_swap:
+            install_python_dev_dependencies.assert_in_venv()
+
+    def test_assert_in_venv_passes_when_in_venv_real_prefix(self) -> None:
+        prefix_swap = self.swap(
+            sys, 'prefix', '/home/user/.pyenv/versions/3.7.10')
+        base_prefix_swap = self.swap(
+            sys, 'base_prefix', '/home/user/.pyenv/versions/3.7.10')
+        real_prefix_swap = self.swap(
+            sys, 'real_prefix', '/home/user/.pyenv/versions/oppia')
+        environ_swap = self.swap(os, 'environ', {})
+        with prefix_swap, base_prefix_swap, real_prefix_swap, environ_swap:
+            install_python_dev_dependencies.assert_in_venv()
+
+    def test_assert_in_venv_fails_when_out_of_venv(self) -> None:
+        prefix_swap = self.swap(
+            sys, 'prefix', '/home/user/.pyenv/versions/3.7.10')
+        base_prefix_swap = self.swap(
+            sys, 'base_prefix', '/home/user/.pyenv/versions/3.7.10')
+        real_prefix_swap = self.swap(sys, 'real_prefix', '')
+        environ_swap = self.swap(os, 'environ', {})
+        expected_error = (
+            'Oppia must be developed within a virtual environment.')
+        with self.assertRaisesRegex(  # type: ignore[no-untyped-call]
+                AssertionError, expected_error):
+            with prefix_swap, base_prefix_swap, real_prefix_swap:
+                with environ_swap:
+                    install_python_dev_dependencies.assert_in_venv()
+
+    def test_assert_in_venv_passes_when_on_ci(self) -> None:
+        prefix_swap = self.swap(
+            sys, 'prefix', '/home/user/.pyenv/versions/3.7.10')
+        base_prefix_swap = self.swap(
+            sys, 'base_prefix', '/home/user/.pyenv/versions/3.7.10')
+        real_prefix_swap = self.swap(sys, 'real_prefix', '')
+        environ_swap = self.swap(os, 'environ', {'GITHUB_ACTION': '1'})
+        with prefix_swap, base_prefix_swap, real_prefix_swap, environ_swap:
+            install_python_dev_dependencies.assert_in_venv()
+
+    def test_install_installation_tools(self) -> None:
+        expected_tools = {
+            'pip': '22.1.1',
+            'pip-tools': '6.6.2',
+            'setuptools': '58.5.3',
+        }
+        installed_tools = {}
+
+        def mock_run(
+            args: List[str], check: bool, encoding: str,
+        ) -> None:
+            package, version = args[-1].split('==')
+            assert package not in installed_tools
```

Comment:
Here and elsewhere: shouldn't you be using self.assertNotIn(), self.assertEquals(), etc.?

Reply:
Ahh yes, I was thinking in pytest mode :)

Reply:
done